packages/wrangler/src/utils/debug-log-file.ts

```diff
@@ -0,0 +1,50 @@
+import path from "node:path";
+import { format } from "node:util";
+import { appendFile } from "node:fs/promises";
+import { getEnvironmentVariableFactory } from "../environment-variables/factory";
+import { getBasePath } from "../paths";
+import { logger } from "../logger";
+
+const getDebugFilepath = getEnvironmentVariableFactory({
+	variableName: "WRANGLER_DEBUG_LOG",
+	defaultValue: () => {
+		const date = new Date()
+			.toISOString()
+			.replaceAll(":", "-")
+			.replace(".", "_")
+			.replace("T", "_")
+			.replace("Z", "");
+		const absoluteFilepath = path.join(
+			getBasePath(),
+			".wrangler",
+			`wrangler-debug-${date}.log`
+		);
+		const relativeFilepath = path.relative(process.cwd(), absoluteFilepath);
+
+		return relativeFilepath;
+	},
+});
+
+const debugLogFilepath = getDebugFilepath();
+
+let hasLoggedLocation = false;
+let hasLoggedError = false;
+
+export function appendToDebugLogFile(...args: unknown[]) {
+	const entry = `
+--- ${new Date().toISOString()}
+${format(...args)}
+---
+`;
+
+	appendFile(debugLogFilepath, entry).catch((err) => {
+		if (hasLoggedError) return;
+		hasLoggedError = true;
+		logger.error(`Failed to write to debug log file`, err);
+	});
+
+	if (hasLoggedLocation) return;
+	hasLoggedLocation = true;
+	logger.info(`🐛 Writing debug logs to "${debugLogFilepath}"`);
+	// TODO: move this into an exit hook so it isn't the first thing logged
```

Comment:
TODO before this PR is merged

Reply:
@admah currently this line is logged upon the first entry to the log file but as mentioned in the comment, it might be better to log this line when the process exits